### PR TITLE
Add Futures to Totals

### PR DESCRIPTION
**Pull request by noahlitvin**
Test deploy: https://thegraph.com/hosted-service/subgraph/noahlitvin/optimism-exchanges?selected=logs

### CODE REVIEW

File: src/exchanges.ts [hunk 1]
```diff
@@ -333,3 +336,39 @@ export function handleFeeChange(event: ExchangeFeeUpdatedEvent): void {
   entity.fee = toDecimal(event.params.newExchangeFeeRate);
   entity.save();
 }
+
+export function handleFuturesPositionModified(event: PositionModifiedEvent): void {
+  let synth = event.transaction.to!.toHex();
```
Comment (artdgn) [inline src/exchanges.ts:341]: nit: `market`?

File: subgraphs/exchanges.js [hunk 2]
```diff
@@ -212,6 +212,83 @@ getContractDeployments('SystemSettings').forEach((a, i) => {
...
+          {
+            event: 'MarketRemoved(address,indexed bytes32,indexed bytes32)',
+            handler: 'handleMarketRemoved',
+          },
+        ],
+      },
+    });
+  });
+
+  const synths = ['BTC', 'ETH', 'LINK'];
```
Comment (artdgn) [inline subgraphs/exchanges.js:256]: This should probably use `getFuturesMarkets` from synthetix package that returns the latest markets (that's e.g. used in the keeper and market closure bots).

Also the entities are not synths, and the keys will not necessarily match existing synths.

**Review by artdgn — APPROVED** [review]
A follow up for the hardcoded markets will be needed.

File: subgraphs/exchanges.js (hunk 2 above)
Comment (noahlitvin) [reply subgraphs/exchanges.js:256]: Updating this to use a template now

File: src/exchanges.ts [hunk 3]
```diff
@@ -337,8 +341,13 @@ export function handleFeeChange(event: ExchangeFeeUpdatedEvent): void {
   entity.save();
 }
 
+export function handleMarketAdded(event: MarketAddedEvent): void {
+  let context = new DataSourceContext();
+  FuturesMarketTemplate.createWithContext(event.params.market, context);
```
Comment (artdgn) [inline src/exchanges.ts]: will there be a problem if a previously existing market is added to a newly deployed manager?

**Review by dbeal-eth — APPROVED** [review]

**Pull request merged**